### ISSUE TITLE
[micromamba] `micromamba env export` does not show packages installed with `pip`

### ISSUE THREAD

**Issue by zacharyburnett** [mamba-org/mamba#2008]
I created an environment with `micromamba`:
```shell
micromamba create -n mamba_export_test
```
and then installed a package with `pip`:
```
micromamba activate mamba_export_test
pip install requests
```
However, when running `micromamba env export`, the packages installed with `pip` are not shown:
```yaml
name: mamba_export_test
channels:
- https://repo.anaconda.com/pkgs/main
dependencies:
- python=3.10.6=hdfd78df_0
- pip=22.2.2=py310hecd8cb5_0
- xz=5.2.6=hca72f7f_0
- ncurses=6.3=hca72f7f_3
- readline=8.1.2=hca72f7f_1
- ca-certificates=2022.07.19=hecd8cb5_0
- zlib=1.2.12=h4dc903c_3
- libcxx=14.0.6=h9765a3e_0
- bzip2=1.0.8=h1de35cc_0
- wheel=0.37.1=pyhd3eb1b0_0
- tk=8.6.12=h5d9f67b_0
- libffi=3.3=hb1e8313_2
- tzdata=2022c=h04d1e81_0
- openssl=1.1.1q=hca72f7f_0
- certifi=2022.9.24=py310hecd8cb5_0
- sqlite=3.39.3=h707629a_0
- setuptools=63.4.1=py310hecd8cb5_0
```
In contrast, `conda env export` shows the `pip` packages:
```yaml
name: /Users/zburnett/micromamba/envs/mamba_export_test
channels:
  - defaults
dependencies:
  - bzip2=1.0.8=h1de35cc_0
  - ca-certificates=2022.07.19=hecd8cb5_0
  - certifi=2022.9.24=py310hecd8cb5_0
  - libcxx=14.0.6=h9765a3e_0
  - libffi=3.3=hb1e8313_2
  - ncurses=6.3=hca72f7f_3
  - openssl=1.1.1q=hca72f7f_0
  - pip=22.2.2=py310hecd8cb5_0
  - python=3.10.6=hdfd78df_0
  - readline=8.1.2=hca72f7f_1
  - setuptools=63.4.1=py310hecd8cb5_0
  - sqlite=3.39.3=h707629a_0
  - tk=8.6.12=h5d9f67b_0
  - tzdata=2022c=h04d1e81_0
  - wheel=0.37.1=pyhd3eb1b0_0
  - xz=5.2.6=hca72f7f_0
  - zlib=1.2.12=h4dc903c_3
  - pip:
    - charset-normalizer==2.1.1
    - idna==3.4
    - requests==2.28.1
    - urllib3==1.26.12
prefix: /Users/zburnett/micromamba/envs/mamba_export_test
```

closed by #2104
closed by #3565

**Comment by jmakov** [comment]
Um... how is this not a priority bug? Am I using (micro)mamba wrong?

**Comment by massich** [comment]
`list`  suffer from the same issue. 

```sh
(prep) sik@dory:/tmp/tmp.OFUuqwduLJ$ micromamba env create -n foo -c conda-forge -q -y python pip
(prep) sik@dory:/tmp/tmp.OFUuqwduLJ$ micromamba activate foo
(foo) sik@dory:/tmp/tmp.OFUuqwduLJ$ pip install requests
Collecting requests
  Using cached requests-2.28.2-py3-none-any.whl (62 kB)
Collecting charset-normalizer<4,>=2
  Downloading charset_normalizer-3.1.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (197 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 197.3/197.3 kB 3.0 MB/s eta 0:00:00
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting urllib3<1.27,>=1.21.1
  Using cached urllib3-1.26.14-py2.py3-none-any.whl (140 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2022.12.7-py3-none-any.whl (155 kB)
Installing collected packages: urllib3, idna, charset-normalizer, certifi, requests
Successfully installed certifi-2022.12.7 charset-normalizer-3.1.0 idna-3.4 requests-2.28.2 urllib3-1.26.14
(foo) sik@dory:/tmp/tmp.OFUuqwduLJ$ micromamba list
List of packages in environment: "/home/sik/opt/conda/envs/foo"

  Name              Version    Build               Channel    
────────────────────────────────────────────────────────────────
  _libgcc_mutex     0.1        conda_forge         conda-forge
  _openmp_mutex     4.5        2_gnu               conda-forge
  bzip2             1.0.8      h7f98852_4          conda-forge
  ca-certificates   2022.12.7  ha878542_0          conda-forge
  ld_impl_linux-64  2.40       h41732ed_0          conda-forge
  libffi            3.4.2      h7f98852_5          conda-forge
  libgcc-ng         12.2.0     h65d4601_19         conda-forge
  libgomp           12.2.0     h65d4601_19         conda-forge
  libnsl            2.0.0      h7f98852_0          conda-forge
  libsqlite         3.40.0     h753d276_0          conda-forge
  libuuid           2.32.1     h7f98852_1000       conda-forge
  libzlib           1.2.13     h166bdaf_4          conda-forge
  ncurses           6.3        h27087fc_1          conda-forge
  openssl           3.0.8      h0b41bf4_0          conda-forge
  pip               23.0.1     pyhd8ed1ab_0        conda-forge
  python            3.11.0     he550d4f_1_cpython  conda-forge
  readline          8.1.2      h0f457ee_0          conda-forge
  setuptools        67.5.1     pyhd8ed1ab_0        conda-forge
  tk                8.6.12     h27826a3_0          conda-forge
  tzdata            2022g      h191b570_0          conda-forge
  wheel             0.38.4     pyhd8ed1ab_0        conda-forge
  xz                5.2.6      h166bdaf_0          conda-forge
(foo) sik@dory:/tmp/tmp.OFUuqwduLJ$ pip list
Package            Version
------------------ ---------
certifi            2022.12.7
charset-normalizer 3.1.0
idna               3.4
pip                23.0.1
requests           2.28.2
setuptools         67.5.1
urllib3            1.26.14
wheel              0.38.4
```

**Comment by zN3utr4l** [comment]
Are there any updates regarding this bug?

**Comment by jonashaag** [comment]
I think this is the latest state https://github.com/mamba-org/mamba/pull/2104

**Comment by jjerphan** [comment]
FYI, https://github.com/mamba-org/mamba/pull/3565 addressed the issue with `list`.

**Comment by michal-kuzak** [comment]
Hi
I'm running the new 2.0.4 version, and yes export shows the pip packages now, but --explicit is still not working

```
(base) mambauser@817e268093b7:/tmp$ micromamba  env export
name: base
channels:
  - conda-forge
dependencies:
  - _libgcc_mutex=0.1=conda_forge
  - _openmp_mutex=4.5=2_gnu
  - bzip2=1.0.8=h4bc722e_7
  - ca-certificates=2024.8.30=hbcca054_0
  - ld_impl_linux-64=2.43=h712a8e2_2
  - libexpat=2.6.4=h5888daf_0
  - libffi=3.4.2=h7f98852_5
  - libgcc=14.2.0=h77fa898_1
  - libgcc-ng=14.2.0=h69a702a_1
  - libgomp=14.2.0=h77fa898_1
  - libmpdec=4.0.0=h4bc722e_0
  - libsqlite=3.47.0=hadc24fc_1
  - libuuid=2.38.1=h0b41bf4_0
  - libzlib=1.3.1=hb9d3cd8_2
  - ncurses=6.5=he02047a_1
  - openssl=3.4.0=hb9d3cd8_0
  - pip=24.3.1=pyh145f28c_0
  - python=3.13.0=h9ebbce0_100_cp313
  - python_abi=3.13=5_cp313
  - readline=8.2=h8228510_1
  - tk=8.6.13=noxft_h4845f30_101
  - tzdata=2024b=hc8b5060_0
  - xz=5.2.6=h166bdaf_0
  - pip:
    - hazelcast==0.1.0
    - hazelcast-python-client==5.5.0
```
```
(base) mambauser@817e268093b7:/tmp$ micromamba  env export --explicit
# This file may be used to create an environment using:
# $ conda create --name <env> --file <this file>
# platform: linux-64
@EXPLICIT
https://conda.anaconda.org/conda-forge/linux-64/_libgcc_mutex-0.1-conda_forge.tar.bz2#d7c89558ba9fa0495403155b64376d81
https://conda.anaconda.org/conda-forge/linux-64/libgomp-14.2.0-h77fa898_1.conda#cc3573974587f12dda90d96e3e55a702
https://conda.anaconda.org/conda-forge/linux-64/_openmp_mutex-4.5-2_gnu.tar.bz2#73aaf86a425cc6e73fcf236a5a46396d
https://conda.anaconda.org/conda-forge/linux-64/libgcc-14.2.0-h77fa898_1.conda#3cb76c3f10d3bc7f1105b2fc9db984df
https://conda.anaconda.org/conda-forge/linux-64/libgcc-ng-14.2.0-h69a702a_1.conda#e39480b9ca41323497b05492a63bc35b
https://conda.anaconda.org/conda-forge/linux-64/bzip2-1.0.8-h4bc722e_7.conda#62ee74e96c5ebb0af99386de58cf9553
https://conda.anaconda.org/conda-forge/linux-64/ca-certificates-2024.8.30-hbcca054_0.conda#c27d1c142233b5bc9ca570c6e2e0c244
https://conda.anaconda.org/conda-forge/linux-64/ld_impl_linux-64-2.43-h712a8e2_2.conda#048b02e3962f066da18efe3a21b77672
https://conda.anaconda.org/conda-forge/linux-64/libexpat-2.6.4-h5888daf_0.conda#db833e03127376d461e1e13e76f09b6c
https://conda.anaconda.org/conda-forge/linux-64/libffi-3.4.2-h7f98852_5.tar.bz2#d645c6d2ac96843a2bfaccd2d62b3ac3
https://conda.anaconda.org/conda-forge/linux-64/libmpdec-4.0.0-h4bc722e_0.conda#aeb98fdeb2e8f25d43ef71fbacbeec80
https://conda.anaconda.org/conda-forge/linux-64/libzlib-1.3.1-hb9d3cd8_2.conda#edb0dca6bc32e4f4789199455a1dbeb8
https://conda.anaconda.org/conda-forge/linux-64/libsqlite-3.47.0-hadc24fc_1.conda#b6f02b52a174e612e89548f4663ce56a
https://conda.anaconda.org/conda-forge/linux-64/libuuid-2.38.1-h0b41bf4_0.conda#40b61aab5c7ba9ff276c41cfffe6b80b
https://conda.anaconda.org/conda-forge/linux-64/ncurses-6.5-he02047a_1.conda#70caf8bb6cf39a0b6b7efc885f51c0fe
https://conda.anaconda.org/conda-forge/linux-64/openssl-3.4.0-hb9d3cd8_0.conda#23cc74f77eb99315c0360ec3533147a9
https://conda.anaconda.org/conda-forge/linux-64/python_abi-3.13-5_cp313.conda#381bbd2a92c863f640a55b6ff3c35161
https://conda.anaconda.org/conda-forge/linux-64/readline-8.2-h8228510_1.conda#47d31b792659ce70f470b5c82fdfb7a4
https://conda.anaconda.org/conda-forge/linux-64/tk-8.6.13-noxft_h4845f30_101.conda#d453b98d9c83e71da0741bb0ff4d76bc
https://conda.anaconda.org/conda-forge/noarch/tzdata-2024b-hc8b5060_0.conda#8ac3367aafb1cc0a068483c580af8015
https://conda.anaconda.org/conda-forge/linux-64/xz-5.2.6-h166bdaf_0.tar.bz2#2161070d867d1b1204ea749c8eec4ef0
https://conda.anaconda.org/conda-forge/linux-64/python-3.13.0-h9ebbce0_100_cp313.conda#08e9aef080f33daeb192b2ddc7e4721f
https://conda.anaconda.org/conda-forge/noarch/pip-24.3.1-pyh145f28c_0.conda#ca3afe2d7b893a8c8cdf489d30a2b1a3
```

**Comment by jjerphan** [comment]
As mentioned in https://github.com/mamba-org/mamba/pull/3623#issue-2676327642, conda does not export wheels in [the explicit environment specification](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#building-identical-conda-environments).

We can consider including wheels from PyPI within the explicit specification.

**Comment by mikekuzak** [comment]
thanks for the clarification. Is there any workaround for this?

**Comment by wookayin** [comment]
TL;DR - this was done in #3623 (released in mamba 2.0.6+).
